Prevent `Update Notebook dts` action from triggering on forked repositories

corbob opened the pull request:
Similar to https://github.com/PowerShell/PowerShell/pull/12763

## PR Summary

Update GitHub Actions Workflow to only trigger on the main vscode-powershell repository.

Note: I have not tested this, merely modeled it after the [documentation](https://docs.github.com/en/actions/reference/workflow-syntax-for-github-actions#example-using-contexts) and the PowerShell PR.

## PR Checklist

- [x] PR has a meaningful title
- [x] Summarized changes
- [NA] PR has tests
- [x] This PR is ready to merge and is not work in progress

TylerLeonhardt commented:
I was kinda hoping this would prevent the entire action from running. In the other PR the `if` is at a higher level.

Can you follow that?

corbob commented:
I've moved the if up to the steps level. I tried it in the on level, but it didn't like that. I also added it to the codeql workflow so it too only runs on the main repository. From my testing it works:
![image](https://user-images.githubusercontent.com/30301021/89567277-3a576280-d7d6-11ea-8384-cba3475bab44.png)
